DD2480-CI/src/main/java/skeleton/TestAutomationHandler.java

```diff
@@ -0,0 +1,64 @@
+package skeleton;
+
+import mavenBuilder.MavenInvokerBuilder;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.json.JSONObject;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+
+public class TestAutomationHandler {
+    String branch;
+    String cloneURL;
+    String[] commitIDs;
+    String sha;
+
+
+    /**
+     * Creates and initializes a TestAutomationHandler object.
+     *
+     * @param request the Github POST request
+     */
+    public TestAutomationHandler(HttpServletRequest request) {
+        RequestParser requestParser = new RequestParser(request);
+        this.branch = requestParser.branch;
+        this.cloneURL = requestParser.cloneURL;
+        this.commitIDs = requestParser.commitIDs;
+        this.sha = this.commitIDs[0];
+    }
+
+    /**
+     * Run the tests from targeted maven directory
+     */
+    public void runTests() {
+        testUpdateStatus(BuildStatus.PENDING);
+        try {
+            GitRepo gitRepo = new GitRepo(cloneURL, branch);
```

Comment:
I think you missed cloning the repo

Reply:
The cloneRepo() is a private function and was called in class initialization.

Reply:
Sorry, yes you are correct!